backend/src/main/java/com/google/sps/data/Participant.java

```diff
@@ -1,27 +1,35 @@
 package com.google.sps.data;
 
+import java.time.ZonedDateTime;
+
 /** A user who want to be matched. */
 public final class Participant {
 
   /** Datastore ID */
   private final long id;
-  /** Google username */
-  private final String ldap;
+  /** Google username (ldap) */
+  private final String username;
+  /** Time user is starts being available */
+  private final ZonedDateTime startTimeAvailable;
   /** Time user is available until */
-  private final long timeAvailableUntil;
-  /** User timezone */
-  private final String timezone;
+  private final ZonedDateTime endTimeAvailable;
   /** How long user wants to chat */
   private final int duration;
   /** Time of submitted form */
   private final long timestamp;
-  
+
   /** Initialize constructor fields */
-  public Participant(long id, String ldap, long timeAvailableUntil, String timezone, int duration, long timestamp) {
+  public Participant(
+      long id,
+      String username,
+      ZonedDateTime startTimeAvailable,
+      ZonedDateTime endTimeAvailable,
+      int duration,
+      long timestamp) {
     this.id = id;
```

Comment:
`startTimeAvailable` and `endTimeAvailable` have an implicit dependency on each other that `endTimeAvailable` is after `startTimeAvailable` but there are no guards to prevent this class from being constructed with date times that don't match this invariant. I would suggest either:

1. Replacing endTimeAvailable with an availability duration. I believe that a ZoneDateTime start and a long duration should be enough to construct a ZoneDateTime for the end. The main downside here is that this class has to worry about two separate durations, one for the availability window length and one for the desired meeting length
2. Add a precondition check (for example with this class https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html) that asserts that start is before after.

Reply:
Added a precondition check to the Participant constructor.